crates/sqlexec/src/remote/planner.rs

```diff
@@ -306,17 +382,30 @@ impl<'a> RemotePhysicalPlanner<'a> {
     /// physical plan.
     fn replace_local_runtime_groups(
         &self,
-        plan: Arc<dyn ExecutionPlan>,
+        original_plan: Arc<dyn ExecutionPlan>,
     ) -> Result<(Arc<dyn ExecutionPlan>, Vec<ClientExchangeSendExec>)> {
         let mut sends = Vec::new();
+        let pref = original_plan
+            .as_any()
+            .downcast_ref::<RuntimeGroupExec>()
+            .map(|exec| exec.preference)
+            .unwrap_or(RuntimePreference::Unspecified);
 
-        let plan = plan.transform_up_mut(&mut |plan| {
+        let plan = original_plan.transform_up_mut(&mut |plan| {
             let mut new_children: Vec<Arc<dyn ExecutionPlan>> = Vec::new();
             let mut did_modify = false;
 
             for child in plan.children() {
                 match child.as_any().downcast_ref::<RuntimeGroupExec>() {
                     Some(exec) if exec.preference == RuntimePreference::Local => {
+                        if matches!(pref, RuntimePreference::Local) {
+                            // If the root of the plan is local, then we can just
+                            // execute everything locally by omitting the remote
+                            // execution exec.
+                            new_children.push(child);
+                            continue;
+                        }
```

Comment:
this was needed for some cases where it was trying to push local execs to remote even though the parent was also a local exec. such as `create temp table t1 as select * from generate_series(...)`

Previously this was handled by the rewriter, but now everything happens in a single pass in the extension planner, so some additional logic was needed in here.